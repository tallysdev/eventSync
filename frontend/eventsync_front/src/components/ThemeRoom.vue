<template>
    <v-main class="h-100 pa-0">
        <v-container class="">
            <h1>Evento: Nome do Evento</h1>
            <h1 class="text-center">Criar sala temática.</h1>
            <v-form class="mb-10">
                <v-text-field variant="outlined" label="Data Inicial" type="date"></v-text-field>
                <v-text-field variant="outlined" label="Data Final" type="date"></v-text-field>
                <v-text-field variant="outlined"
                name="name"
                label="Horario inicial da sala."
                id="id"
                type="time"
                ></v-text-field>
                <v-text-field variant="outlined" label="Local da sala"></v-text-field>
                <v-text-field variant="outlined" label="Número Min. de Participantes"></v-text-field>
                <v-text-field variant="outlined" label="Número Máx. de Participantes"></v-text-field>
                <v-text-field variant="outlined" label="Palestrante"></v-text-field>
                <v-text-field variant="outlined" label="Nome da sala"></v-text-field>
                <v-text-field variant="outlined" label="Descrição"></v-text-field>
                <v-text-field variant="outlined" label="Quantidade horas no certificado"></v-text-field>
                <v-text-field variant="outlined" label="Público Alvo"></v-text-field>
                <v-text-field variant="outlined" label="Tipo de evento."></v-text-field>

                <v-container grid-list-xs class="d-flex justify-center ga-10">
                    <v-btn color="primary">Criar</v-btn>
                    <v-btn color="secondary">Voltar</v-btn>
                </v-container>
            </v-form>
        </v-container>
    </v-main>
</template>

<script>
export default {

}
</script>
<style lang="">

</style>