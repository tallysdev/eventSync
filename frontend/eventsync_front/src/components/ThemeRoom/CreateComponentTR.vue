<template>
    <v-main class="h-100 pa-0">
        <v-container>
            <h1>Evento: Nome do Evento</h1>
            <h1 class="text-center">Criar sala temática.</h1>
            <v-form 
            v-model="valid"
            ref="form"
            lazy-validation
            @submit.prevent="submitForm" 
            class="mb-10">

                <v-text-field 
                v-model="roomForm.name" 
                variant="outlined" 
                label="Nome da sala" 
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.start_date" 
                variant="outlined" 
                label="Data Inicial" 
                type="date"
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.end_date" 
                variant="outlined" 
                label="Data Final" 
                type="date"
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.start_time" 
                variant="outlined" 
                label="Horário inicial da sala" 
                type="time"
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.local" 
                variant="outlined" 
                label="Local da sala" 
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.min_quantity" 
                variant="outlined" 
                label="Número Min. de Participantes"
                type="number"
                min="1"
                @keypress="validateNumberInput"
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.max_quantity" 
                variant="outlined" 
                label="Número Máx. de Participantes"
                type="number"
                min="1"
                @keypress="validateNumberInput"
                required></v-text-field>

                <v-text-field 
                v-model="roomForm.speaker" 
                variant="outlined" 
                label="Palestrante" 
                required></v-text-field>

                <v-textarea 
                v-model="roomForm.description" 
                variant="outlined" 
                label="Descrição" 
                required></v-textarea>

                <v-text-field 
                v-model="roomForm.hours_quantity" 
                variant="outlined"
                min="1"
                @keypress="validateNumberInput"
                label="Quantidade horas no certificado" 
                type="number"></v-text-field>

                <v-text-field 
                v-model="roomForm.audiences"
                variant="outlined"
                label="Público Alvo" 
                required></v-text-field>

                <v-select
                v-model="roomForm.event_type"
                :items="eventTypeOptions" 
                variant="outlined" 
                label="Tipo de evento"
                required></v-select>

                <v-container grid-list-xs class="d-flex justify-center ga-10">
                    <v-btn 
                    type="submit" 
                    color="primary"
                    :disabled="!isFormValid || submitting"
                    >Criar</v-btn>
                    <v-btn color="secondary" @click="goBack">Voltar</v-btn>
                </v-container>
            </v-form>

            <!-- Snackbar para mensagens de sucesso ou erro -->
            <v-snackbar v-model="snackbar" :color="snackbarColor" timeout="3000" top right>
                {{ snackbarText }}
                <template v-slot:action="{ attrs }">
                    <v-btn color="white" text v-bind="attrs" @click="snackbar = false">Fechar</v-btn>
                </template>
            </v-snackbar>
        </v-container>
    </v-main>
</template>


<script setup lang="ts">
import { ref, computed } from 'vue';
import { useRouter } from 'vue-router';
import { addThemeRoom, updateThemeRoom } from '@/services/themRoomService';
import { validateFields, validateNumberInput, snackbar, snackbarText, snackbarColor } from '../../stores/validatorRoom';

// Classe do formulário
class ThemeRoomForm {
    name = "";
    start_time = "";
    start_date = "";
    end_date = "";
    local = "";
    min_quantity = '';
    max_quantity = '';
    speaker = "";
    description = "";
    hours_quantity = '';
    audiences = "";
    event_type = "";
    status = "upcoming";
}

const roomForm = ref(new ThemeRoomForm());
const valid = ref(false);
const submitting = ref(false);
const errorMessage = ref<string | null>(null);
const showError = ref(false);
const eventTypeOptions = ['conference', 'workshop', 'seminar', 'meetup'];

const router = useRouter();

const isFormValid = computed(() => {
    return Object.values(roomForm.value).every(field => field);
});

const showSnackbar = (message: string, type: string) => {
    snackbarText.value = message;
    snackbarColor.value = type === 'success' ? 'green' : 'red';
    snackbar.value = true;
};

const resetForm = () => {
    roomForm.value = new ThemeRoomForm();
};

const buildFormData = () => {
    const form = new FormData();
    Object.entries(roomForm.value).forEach(([key, value]) => {
        form.append(key, value.toString());
        console.log(key, value);
    });
    form.append('event_id', router.currentRoute.value.params.id.toString());
    return form;
};

const addRoom = async (form: FormData) => {
    await addThemeRoom(form);
    showSnackbar('Sala criada com sucesso', 'success');
};

const updateRoom = async (roomId: number, form: FormData) => {
    await updateThemeRoom(roomId, form);
    showSnackbar('Sala atualizada com sucesso', 'success');
};

const submitForm = async () => {
    if (submitting.value || !isFormValid.value) return;

    submitting.value = true;

    if (!validateFields(roomForm.value)) {
        errorMessage.value = 'Por favor, preencha todos os campos obrigatórios.';
        submitting.value = false;
        return;
    }

    const form = buildFormData();

    try {
        const roomId = router.currentRoute.value.params.room_id;
        if (roomId) {
            await updateRoom(Number(roomId), form);
        } else {
            await addRoom(form);
        }
        
        resetForm();
        router.back();

    } catch (error) {
        console.error('Erro ao processar sala:', error);
        showSnackbar(error.response.data.detail.toString() ?? 'Erro ao processar sala', 'error');
        errorMessage.value = 'Erro ao processar sala. Por favor, tente novamente.';
        showError.value = true;
    } finally {
        submitting.value = false;
    }
};

const goBack = () => {
    router.back();
};
</script>


<style scoped></style>
