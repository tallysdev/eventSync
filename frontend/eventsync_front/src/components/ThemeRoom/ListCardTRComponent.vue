<template>
    <v-container fluid class="ma-4 d-flex ga-5 w-100 flex-wrap border-red">
        <v-card class="mx-auto flex-wrap" variant="outlined" max-width="400" v-for="(room, id) in rooms" :key="id">
            <v-card-title class="font-weight-bold text-h5">{{ room.name }}</v-card-title>

            <v-card-subtitle class="text-caption grey--text">
                <v-icon class="mr-1" small>mdi-calendar</v-icon>
                De {{ room.start_date }} - Até {{ room.end_date }}
            </v-card-subtitle>

            <v-card-subtitle class="text-caption grey--text">
                <v-icon class="mr-1" small>mdi-clock-outline</v-icon>
                Horário de início {{ room.start_time }}
            </v-card-subtitle>

            <v-card-text>
                <v-row>
                    <v-col cols="12">
                        <v-icon class="mr-1" small>mdi-map-marker-outline</v-icon>
                        <span class="font-weight-bold">{{ room.local }}</span>
                    </v-col>

                    <v-col cols="12">
                        <v-icon class="mr-1" small>mdi-account-group</v-icon>
                        <span class="font-weight-bold">Participantes: </span>
                        Máximo {{ room.max_quantity }} - Mínimo {{ room.min_quantity }}
                    </v-col>

                    <v-col cols="12">
                        <v-icon class="mr-1" small>mdi-information-outline</v-icon>
                        <span class="font-weight-bold">Tipo: </span>
                        {{ room.event_type }}
                    </v-col>

                    <v-col cols="12">
                        <p class="mt-2">{{ room.description }}</p>
                    </v-col>

                    <v-col cols="12">
                        <p>Público alvo: {{ room.audiences }}</p>
                    </v-col>

                    <v-col cols="12">
                        <p>Palestrante: {{ room.speaker }}</p>
                    </v-col>
                </v-row>
            </v-card-text>

            <v-card-actions>
                <v-btn color="blue" text @click="toggleDetails(id)">Ver Detalhes</v-btn>
                <v-btn color="green" text>Participar</v-btn>
            </v-card-actions>

            <v-expand-transition>
                <div v-show="room.showDetails">
                    <v-divider></v-divider>

                    <v-card-text>
                        Detalhes adicionais da Lorem ipsum dolor sit amet consectetur adipisicing elit. Placeat neque pariatur veniam, distinctio illo dolores, quisquam molestiae iure voluptates ab consequuntur, quis suscipit autem qui fuga facilis facere aliquid. Molestias. sala. Você pode adicionar mais informações aqui.
                    </v-card-text>
                </div>
            </v-expand-transition>
        </v-card>
    </v-container>
</template>


<script setup lang="ts">
import { ref, onMounted, watch } from 'vue'
import { fetchThemeRooms } from '@/services/themRoomService';
import { type ThemeRoom } from '@/types/themeRoom';

const rooms = ref<ThemeRoom[]>([])
const currentPage = ref(1)
const totalPages = ref(1)
const itemsPerPage = 12
const loading = ref(false)
const errorMessage = ref<string | null>(null)

// Função para alternar o estado 'showDetails' para cada sala individualmente
const toggleDetails = (id: string) => {
  const room = rooms.value.find((room: { id: string; }) => room.id === id)
  if (room) {
    room.showDetails = !room.showDetails
  }
}

const fetchRoomsData = async () => {
  loading.value = true
  errorMessage.value = null

  const loadingTimeout = setTimeout(() => {
    loading.value = false
    errorMessage.value = 'O tempo de carregamento expirou. Tente novamente.'
  }, 10000) // 10 seconds timeout

  try {
    const response = await fetchThemeRooms(currentPage.value, itemsPerPage)
    rooms.value = response.data.results.map((room: ThemeRoom) => ({
      ...room,
      showDetails: false, // Adiciona a propriedade showDetails a cada sala
    }))
    console.log(rooms.value)
    totalPages.value = Math.ceil(response.data.count / itemsPerPage)
    clearTimeout(loadingTimeout)
  } catch (error) {
    console.error('Error fetching rooms:', error)
  } finally {
    loading.value = false
  }
}

onMounted(() => {
  fetchRoomsData()
})

watch(currentPage, () => {
  fetchRoomsData()
})
</script>

<style scoped lang="css"></style>
<style scoped lang="css"></style>