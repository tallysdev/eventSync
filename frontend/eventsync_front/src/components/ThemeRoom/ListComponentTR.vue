<template>
    <NavBar />
    <v-main>
        <h1 class="text-center">Evento {{ EventId }}</h1>
        <h1 class="mx-10">Programação:</h1>
        <v-container fluid class="d-flex flex-column align-center">
            <v-btn 
            :to="`/events/${EventId}/create/theme-room`" 
            color="success"
            >Cadastrar Sala Tematica
            </v-btn>
            <h2 v-if="rooms.length === 0">Nenhuma Sala Cadastrada ainda...</h2>
            <v-container fluid class="d-flex flex-wrap ga-5 justify-center">
                <v-card class="mx-2 my-2 justify-center w-75" variant="outlined" elevation="20" max-width="400"
                    min-width="50" v-for="(room, id) in rooms" :key="id">
                    <v-card-actions v-if="admin" class="d-flex flex-wrap w-100 justify-space-between">
                        <v-btn color="yellow" variant="text" :to="`/events/${EventId}/edit/theme-room/${room.id}`">Editar</v-btn>
                        <v-btn color="red" variant="text" @click="fetchDeleteRoom(room.id)">Exluir</v-btn>
                    </v-card-actions>
                    <v-card-title class="font-weight-bold text-h5 text-center">{{ room.name }}</v-card-title>
                    <v-card-subtitle class="text-caption grey--text">
                        <v-icon class="mr-1" small>mdi-calendar</v-icon>
                        De {{ room.start_date }} - Até {{ room.end_date }}
                    </v-card-subtitle>
                    <v-card-subtitle class="text-caption grey--text">
                        <v-icon class="mr-1" small>mdi-clock-outline</v-icon>
                        Horário de início {{ room.start_time }}
                    </v-card-subtitle>
                    <v-card-subtitle class="text-caption grey--text">
                        <v-icon class="mr-1" small>mdi-map-marker-outline</v-icon>
                        <span class="font-weight-bold">{{ room.local }}</span>
                    </v-card-subtitle>
                    <v-card-text>
                        <v-row class="d-flex flex-column">
                            <v-col>
                                <p class="font-weight-bold">Palestrante: {{ room.speaker }}</p>
                            </v-col>
                            <v-col>
                                <v-icon class="mr-1" small>mdi-information-outline</v-icon>
                                <span class="font-weight-bold">Tipo: </span>
                                {{ room.event_type }}
                            </v-col>
                            <v-col class="d-flex flex-wrap">
                                <v-icon class="mr-1" small>mdi-account-group</v-icon>
                                <p class="font-weight-bold">Participantes:</p>
                                <p>Máximo {{ room.max_quantity }} - Mínimo {{ room.min_quantity }}
                                </p>
                            </v-col>
                        </v-row>
                    </v-card-text>
                    <v-card-actions class="d-flex flex-wrap w-100 justify-center">
                        <v-btn color="blue" variant="text" @click="toggleDetails(room.id)">Ver Detalhes</v-btn>
                        <v-btn color="green" variant="text">Participar</v-btn>
                    </v-card-actions>
                    <v-expand-transition>
                        <div v-show="room.showDetails">
                            <v-divider></v-divider>
                            <v-row class="pa-2">
                                <v-card-text>
                                    Público Alvo: {{ room.audiences }}
                                </v-card-text>
                                <v-card-text>
                                    Descrição: {{ room.description }}
                                </v-card-text>
                            </v-row>
                        </div>
                    </v-expand-transition>
                </v-card>
            </v-container>
        </v-container>
    </v-main>
    <FooterVue />
</template>

<script setup lang="ts">
import NavBar from '@/components/NavBar.vue'
import FooterVue from '@/components/Footer.vue'
import { ref, onMounted } from 'vue'
import { fetchThemeRooms, deleteThemeRoom } from '@/services/themRoomService';
import { type ThemeRoom } from '@/types/themeRoom';
import { useRoute } from 'vue-router';

const route = useRoute()
const EventId = Number(route.params.id)
const rooms = ref<ThemeRoom[]>([])
const loading = ref(false)
const errorMessage = ref<string | null>(null)
const admin = true

// Função para alternar o estado 'showDetails' para cada sala individualmente
const toggleDetails = (id: number) => {

    const room = rooms.value.find((room: { id: number; }) => room.id === id)
    if (room) {
        room.showDetails = !room.showDetails
    }
}

const fetchRoomsData = async () => {
    loading.value = true
    errorMessage.value = null

    const loadingTimeout = setTimeout(() => {
        loading.value = false
        errorMessage.value = 'O tempo de carregamento expirou. Tente novamente.'
    }, 10000) // 10 seconds timeout

    try {
        const response = await fetchThemeRooms(EventId)
        // rooms.value = response.data
        rooms.value = response.data.map((room: ThemeRoom) => ({
            ...room,
            showDetails: false, // Adiciona a propriedade showDetails a cada sala
        }))
        clearTimeout(loadingTimeout)
    } catch (error) {
        console.error('Error fetching rooms:', error)
    } finally {
        loading.value = false
    }
}

const fetchDeleteRoom = async (id: number) => {
    loading.value = true
    errorMessage.value = null

    const loadingTimeout = setTimeout(() => {
        loading.value = false
        errorMessage.value = 'O tempo de carregamento expirou. Tente novamente.'
    }, 10000) // 10 seconds timeout

    try {
        const response = await deleteThemeRoom(id)
        clearTimeout(loadingTimeout)
    } catch (error) {
        console.error('Error deleting room:', error)
    } finally {
        loading.value = false
        fetchRoomsData()
    }
}

onMounted(() => {
    fetchRoomsData()
})

</script>

<style scoped lang="css"></style>