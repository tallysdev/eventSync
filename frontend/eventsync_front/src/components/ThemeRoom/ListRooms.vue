<template>
<v-container fluid class="d-flex flex-column align-center">
    <v-btn v-if="admin" :to="`/events/${EventId}/create/theme-room`" color="success">Cadastrar Sala Tematica
    </v-btn>
    <h2 v-if="rooms.length === 0">Nenhuma Sala Cadastrada ainda...</h2>
    <v-container fluid class="d-flex flex-wrap ga-5 justify-center">
        <v-card class="mx-2 my-2 justify-center w-100" variant="outlined" elevation="20" max-width="400"
            min-width="50" v-for="(room, id) in rooms" :key="id">
            <v-card-actions v-if="admin" class="d-flex flex-wrap w-100 justify-space-between">
                <v-btn color="yellow" variant="text"
                    :to="`/events/${EventId}/edit/theme-room/${room.id}`">Editar</v-btn>
                <v-btn color="red" variant="text" @click="fetchDeleteRoom(room.id)">Exluir</v-btn>
            </v-card-actions>
            <v-card-title class="font-weight-bold text-h5 text-center">{{ room.name }}</v-card-title>
            <v-card-subtitle class="text-caption grey--text">
                <v-icon class="mr-1" small>mdi-calendar</v-icon>
                De {{ room.start_date }} - Até {{ room.end_date }}
            </v-card-subtitle>
            <v-card-subtitle class="text-caption grey--text">
                <v-icon class="mr-1" small>mdi-clock-outline</v-icon>
                Horário de início {{ room.start_time }}
            </v-card-subtitle>
            <v-card-subtitle class="text-caption grey--text">
                <v-icon class="mr-1" small>mdi-map-marker-outline</v-icon>
                <span class="font-weight-bold">{{ room.local }}</span>
            </v-card-subtitle>
            <v-card-text>
                <v-row class="d-flex flex-column">
                    <v-col>
                        <p class="font-weight-bold">Palestrante: {{ room.speaker }}</p>
                    </v-col>
                    <v-col>
                        <v-icon class="mr-1" small>mdi-information-outline</v-icon>
                        <span class="font-weight-bold">Tipo: </span>
                        {{ room.event_type }}
                    </v-col>
                    <v-col class="d-flex flex-wrap">
                        <v-icon class="mr-1" small>mdi-account-group</v-icon>
                        <p class="font-weight-bold">Participantes:</p>
                        <p>Máximo {{ room.max_quantity }} - Mínimo {{ room.min_quantity }}
                        </p>
                    </v-col>
                </v-row>
            </v-card-text>
            <v-card-actions class="d-flex flex-wrap w-100 justify-space-between">
                <v-btn color="blue" text @click="room.showDetails = !room.showDetails">Ver detalhes</v-btn>
                <template v-if="isSubscribed">
                    <v-btn @click="handleUnsubscription" color="error" class="text-no-wrap">
                        <b>Cancelar Inscrição</b>
                    </v-btn>
                </template>
                <v-btn v-else @click="handleSubscription" color="success" class="text-no-wrap">
                    Participar
                </v-btn>
            </v-card-actions>
            <v-expand-transition>
                <div v-show="room.showDetails">
                    <v-divider></v-divider>
                    <v-row class="pa-2">
                        <v-card-text>
                            Público Alvo: {{ room.audiences }}
                        </v-card-text>
                        <v-card-text>
                            Descrição: {{ room.description }}
                        </v-card-text>
                    </v-row>
                </div>
            </v-expand-transition>
        </v-card>
        <v-snackbar v-model="showSnackbar" :color="snackbarColor" timeout="4000" top>
            {{ snackbarMessage }}
        </v-snackbar>
    </v-container>
</v-container>
</template>

<script setup lang="ts">

import { ref, onMounted } from 'vue'
import { fetchThemeRooms, deleteThemeRoom } from '@/services/themRoomService';
import { useAuthStore } from '@/stores/auth';
import {
    signupForEvent,
    cancelSubscription,
    checkUserSubscription
} from '@/services/eventService';
import { type ThemeRoom } from '@/types/themeRoom';
import { useRoute, useRouter } from 'vue-router';

const route = useRoute()
const router = useRouter()
const EventId = Number(route.params.id)
const rooms = ref<ThemeRoom[]>([])
const loading = ref(false)
const errorMessage = ref<string | null>(null)
const admin = true
const authStore = useAuthStore()
const isSubscribed = ref(false)
const showSnackbar = ref(false)
const snackbarMessage = ref('')
const snackbarColor = ref('')

const checkSubscriptionStatus = async () => {
    if (authStore.isAuthenticated) {
        try {
            console.log(authStore.token)
            const response = await checkUserSubscription(rooms.value.id, authStore.getUser.id)
            console.log('Subscription status:', response.data)
            isSubscribed.value = await checkUserSubscription(rooms.value.id, authStore.getUser.id)
        } catch (error) {
            console.error('Error checking subscription status:', error)
        }
    }
}

const handleSubscription = async () => {
    if (authStore.isAuthenticated) {
        try {
            await signupForEvent(rooms.value.id, authStore.getUser.id)
            snackbarMessage.value = 'Inscrição realizada com sucesso!'
            snackbarColor.value = 'success'
            isSubscribed.value = true
        } catch (error) {
            console.error('Error subscribing:', error)
        }
    } else {
        snackbarMessage.value = 'Você precisa estar logado para se inscrever.'
        snackbarColor.value = 'error'
        setTimeout(() => {
            router.push('/login')
        }, 2000)
    }
}

// Função para alternar o estado 'showDetails' para cada sala individualmente
const toggleDetails = (id: number) => {

    const room = rooms.value.find((room: { id: number; }) => room.id === id)
    if (room) {
        room.showDetails = !room.showDetails
    }
}

const fetchRoomsData = async () => {
    loading.value = true
    errorMessage.value = null

    const loadingTimeout = setTimeout(() => {
        loading.value = false
        errorMessage.value = 'O tempo de carregamento expirou. Tente novamente.'
    }, 10000) // 10 seconds timeout

    try {
        const response = await fetchThemeRooms(EventId)
        // rooms.value = response.data
        rooms.value = response.data.map((room: ThemeRoom) => ({
            ...room,
            showDetails: false, // Adiciona a propriedade showDetails a cada sala
        }))
        clearTimeout(loadingTimeout)
    } catch (error) {
        console.error('Error fetching rooms:', error)
    } finally {
        loading.value = false
    }
}

const fetchDeleteRoom = async (id: number) => {
    loading.value = true
    errorMessage.value = null

    const loadingTimeout = setTimeout(() => {
        loading.value = false
        errorMessage.value = 'O tempo de carregamento expirou. Tente novamente.'
    }, 10000) // 10 seconds timeout

    try {
        const response = await deleteThemeRoom(id)
        clearTimeout(loadingTimeout)
    } catch (error) {
        console.error('Error deleting room:', error)
    } finally {
        loading.value = false
        fetchRoomsData()
    }
}

onMounted(() => {
    fetchRoomsData()
})

</script>

<style scoped lang="css"></style>