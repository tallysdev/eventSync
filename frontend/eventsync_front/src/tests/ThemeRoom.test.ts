import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import ThemeRoom from "@/components/ThemeRoom/CreateComponentTR.vue";

describe("ThemeRoom.vue", () => {
    const wrapper = mount(ThemeRoom);

    it("renders correctly", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('contains the event title', () => {
        const title = wrapper.find('h1');
        expect(title.exists()).toBe(true);
        expect(title.text()).toBe('Evento: Nome do Evento');
    });

    it('contains the form title', () => {
        const formTitle = wrapper.find('h1.text-center');
        expect(formTitle.exists()).toBe(true);
        expect(formTitle.text()).toBe('Criar sala temática.');
    });

    it('contains all form fields', () => {
        const labels = [
            'Local da sala',
            'Nome da sala',
            'Descrição',
            'Público Alvo',
            'Data',
            'Horário',
            'Número Min. de Participantes',
            'Número Máx. de Participantes'
        ];

        labels.forEach(label => {
            expect(wrapper.find(`input[label="${label}"]`).exists())
        });
    });

    it('contains create and back buttons', () => {
        const createButton = wrapper.find('v-btn[color="primary"]');
        const backButton = wrapper.find('v-btn[color="secondary"]');

        expect(createButton.exists()).toBe(true);
        expect(createButton.text()).toBe('Criar');

        expect(backButton.exists()).toBe(true);
        expect(backButton.text()).toBe('Voltar');
    });
})