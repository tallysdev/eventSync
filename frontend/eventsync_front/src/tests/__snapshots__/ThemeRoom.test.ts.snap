// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeRoom.vue > renders correctly 1`] = `
"<v-main class="h-100 pa-0">
  <v-container class="">
    <h1>Evento: Nome do Evento</h1>
    <h1 class="text-center">Criar sala temática.</h1>
    <v-form class="mb-10">
      <v-text-field label="Local da sala"></v-text-field>
      <v-text-field label="Nome da sala"></v-text-field>
      <v-text-field label="Descrição"></v-text-field>
      <v-text-field label="Público Alvo"></v-text-field>
      <v-text-field label="Data" type="date"></v-text-field>
      <v-text-field label="Horário"></v-text-field>
      <v-text-field label="Número Min. de Participantes"></v-text-field>
      <v-text-field label="Número Máx. de Participantes"></v-text-field>
      <v-container grid-list-xs="" class="d-flex justify-center ga-10">
        <v-btn color="primary">Criar</v-btn>
        <v-btn color="secondary">Voltar</v-btn>
      </v-container>
    </v-form>
  </v-container>
</v-main>"
`;
